bindings/rust/s2n-tls/src/connection.rs

```diff
@@ -318,6 +318,46 @@ impl Connection {
         Ok(self)
     }
 
+    /// Sets the callback to use for verifying that a hostname from an X.509 certificate is
+    /// trusted.
+    ///
+    /// The default behavior is to require that the hostname match the server name set with
+    /// [`Self::set_server_name()`]. This will likely lead to all client certificates being
+    /// rejected, so the callback will need to be overridden when using client authentication.
+    ///
+    /// The callback may be called more than once during certificate validation as each SAN on
+    /// the certificate will be checked.
```

Comment:
We shouldn't be duplicating our documentation for the Rust bindings. Should we link to the C api documentation instead?

Reply:
The nice thing about including them here is they show up in-line with the [generated docs](https://docs.rs/s2n-tls/latest/s2n_tls/config/struct.Builder.html#method.set_verify_host_callback) and in rust-analyzer.

Reply:
Yep, its unfortunate that we need to duplicate this but if folks are using the bindings then they are also probably using the Rust docs. Linking would be better from development but worse from a user experience since the Rust docs so nice :)

Reply:
It's not better from a user experience if they're not correct / out of date, no matter now nicely formatted they are. I didn't mind how openssl linked their documentation: https://docs.rs/openssl/latest/openssl/ec/struct.EcKey.html They basically add a "this corresponds to X" with a link to the original documentation.

Reply:
Fair point. I can add a link to the C docs and leave some of this duplication out.